Use separate layout for the gallery page

dulldesk opened the pull request.

dulldesk committed:
Rename the gallery layout to gallery-display

To prevent potential mix-up with the gallery page

dulldesk committed:
Create and use the info layout

Jwiggiff reviewed (changes requested):
Is there a specific reason the gallery layout was renamed to gallery-display? I think gallery would be fine. Also I'm a bit hesitant on having the title for all of the gallery pages just be gallery instead of Music Gallery or Dance Gallery etc.

dulldesk committed:
Differentiate title from heading

The page heading will use the "heading" variable if present, and the "title" variable by default.

dulldesk commented:
I renamed the layout to `gallery-display` as the main gallery page has a file name `gallery.html`, I feared that the two files having the same file name may get confusing at some point.
2nd point: fixed in the most recent commit b145819b8d9d6476161ad7386b6348c4414259a0
